src/python/bcc/__init__.py

```diff
@@ -1093,7 +1093,12 @@ def cleanup(self):
             # non-string keys here include the perf_events reader
             if isinstance(k, str):
```

Comment:
I'm not entirely convinced, unless we deprecate (:() the feature that a kprobe event can be polled the same as a perf buffer. Just as a historical point, polling kprobes used to be the only way to get callchains into userspace, all the other infra (perf_submit, stackid table) came several major versions later.